eng/scripts/Language-Settings.ps1

```diff
@@ -172,58 +172,97 @@ function Get-javascript-GithubIoDocIndex()
   GenerateDocfxTocContent -tocContent $tocContent -lang "JavaScript"
 }
 
-# Updates a js CI configuration json.
-# For "latest", we simply set a target package name
-# For "preview", we add @next to the target package name
-function Update-javascript-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId = $null)
-{
-  $pkgJsonLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
-
-  if (-not (Test-Path $pkgJsonLoc))
-  {
-    Write-Error "Unable to locate package json at location $pkgJsonLoc, exiting."
-    exit(1)
+# "@azure/package-name@1.2.3" -> "@azure/package-name"
+function Get-PackageNameFromDocsMsConfig($DocsConfigName) { 
+  if ($DocsConfigName -match '^(@?.+?)(@(.*))?$') { 
+    return $Matches[1]
   }
 
-  $allJson = Get-Content $pkgJsonLoc | ConvertFrom-Json
+  throw "Cannot find package $DocsConfigName"
+}
 
-  $visibleInCI = @{}
+function Update-javascript-DocsMsPackages($DocsRepoLocation)
+{
+  $publishedPackages = Get-CSVMetadata `
+  | Where-Object { $_.New -eq 'true' -and $_.Hide -ne 'true' }
+
+  # Update "Latest" packages
+  $latestConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-latest.json' -Resolve
+  Write-Host "Updating latest configuration: $latestConfigPath"
+  $latestPackages = Get-Content $latestConfigPath -Raw | ConvertFrom-Json
+
+  $gaPackages = $publishedPackages | Where-Object { $_.VersionGA.trim() }
+  foreach ($package in $gaPackages) { 
+    if ($latestPackages.npm_package_sources.name -contains $package.Package) { 
+      # Onboarded packages already contains this package, skip
+      Write-Host "Keeping onboarded package $($package.Package)"
```

Comment:
Do we need to potentially update the version?

Reply:
Version isn't required for latest packages. Pinning the version here could make the builds more deterministic. If the docindex pipeline fails for whatever reason then package versions in `latest` will not update. 

Seems worth the potential downsides to make the builds more deterministic.

Reply:
I was also thinking about this pattern for the other languages that will need a version update in this step. Also we will end up needing to set the version when we add the daily builds to this logic.